### DISCUSSION TITLE
Precision Issue with Database Field

### DISCUSSION THREAD

**Discussion by panxiaole** [teslamate-org/teslamate#4127] [Ideas]
The current type of the 'battery_level' field in the 'position' table is int2. I want to use this field to display energy consumption during driving, but it seems that using an integer type will lose a lot of precision. Can this field be changed to a floating-point type?

**Comment by brianmay** [comment]
I think that that the battery_level is supplied from the Tesla API is an integer value, in which case changing the database field is not going to help.

**Reply by panxiaole** [reply]
OK,I see.
